component/org.wso2.carbon.identity.api.otp.service.emailotp/src/main/java/org/wso2/carbon/identity/api/otp/service/emailotp/impl/EmailotpApiServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.otp.service.emailotp.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.extension.identity.emailotp.common.dto.FailureReasonDTO;
+import org.wso2.carbon.extension.identity.emailotp.common.dto.GenerationResponseDTO;
+import org.wso2.carbon.extension.identity.emailotp.common.dto.ValidationResponseDTO;
+import org.wso2.carbon.extension.identity.emailotp.common.exception.EmailOtpClientException;
+import org.wso2.carbon.extension.identity.emailotp.common.exception.EmailOtpException;
+import org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApiService;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPGenerateResponse;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPGenerationRequest;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPValidationFailureReason;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPValidationRequest;
+import org.wso2.carbon.identity.api.otp.service.emailotp.dto.OTPValidationResponse;
+import org.wso2.carbon.identity.api.otp.service.emailotp.util.EndpointUtils;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * This class implements the service layer for
+ * org.wso2.carbon.identity.api.otp.service.emailotp.EmailotpApi.
+ */
```

Comment:
Let's follow javadoc guidelines for comments. 
For an example, we can use ` {@link EmailotpApi} ` here